### PR TITLE
Database stream schema

### PR DESCRIPTION
**Pull request by sharminelow**
#### Basic Storage Stream API
- Create, Get, Update Stream functions
- Corresponding test cases

### CODE REVIEW

File: app_server/app/models/Storage.js [hunk 1]
```diff
@@ -14,12 +18,11 @@ var _ = require('underscore');
  * @constructor
  */
 function Storage() {
-  var config = rfr('config/DatabaseConfig');
-  this.models = {};
-  this.Sequelize = require('sequelize');
+  var models = {};
+  var Sequelize = require('sequelize');
```
Comment (ndt93) [inline app_server/app/models/Storage.js]: This looks nice to be required at the beginning

File: app_server/app/models/Storage.js [hunk 2]
```diff
@@ -214,20 +229,79 @@ Class.getListOfUsers = function() {
...
+  var streamPromise = this.models.Stream.create(streamAttributes);
+  var that = this;
+
+  return Promise.join(userPromise, streamPromise,
+    function(user, stream) {
+      return user.addStream(stream).then(function() {
+        return that.getStreamById(stream.streamId);
+      });
+    });
+};
```
Comment (ndt93) [inline app_server/app/models/Storage.js:249]: Should be nice to use `bind` there instead of `that`

Comment (ndt93) [inline app_server/app/models/Storage.js:249]: Also remember to implement `catch` clause somewhere in the pipeline. Either here or in upper service layer.

Comment (sharminelow) [inline app_server/app/models/Storage.js:249]: Was reading this https://github.com/petkaantonov/bluebird/issues/92 for the usage of bind() with Promise.all or Promise.join. Seems like I can't bind(this). I managed to get around the issue without binding.

**Comment by ndt93** [comment]
Ok. Good to merge.

**Pull request merged**